src/mini_trainer/setup_model_for_training.py

```diff
@@ -174,11 +187,28 @@ def setup_model(
     base_model_args = {
         "pretrained_model_name_or_path": model_name_or_path,
     }
-    # Check if flash_attn is available, otherwise use eager
-    # in practice we will need flash attention when running this repo
+    
+    # Get model config to check for GPT-OSS and set appropriate configurations
+    model_config = AutoConfig.from_pretrained(model_name_or_path)
+    is_gpt_oss = model_config.model_type == "gpt_oss"
+    
+    # Set up quantization config for GPT-OSS models
+    if is_gpt_oss:
+        try:
+            from transformers import Mxfp4Config
+            base_model_args["quantization_config"] = Mxfp4Config(dequantize=True)
+            log_rank_0("🎯 Detected GPT-OSS model - applying dequantization for training")
+        except ImportError:
+            log_rank_0("⚠️ GPT-OSS model detected but Mxfp4Config not available - using default config")
+    
+    # Check if flash_attn is available and set appropriate attention implementation
     try:
         import flash_attn
-        base_model_args["attn_implementation"] = "flash_attention_2"
+        if is_gpt_oss:
+            base_model_args["attn_implementation"] = "kernels-community/vllm-flash-attn3"
```

Comment:
@Maxusmusti I saw in the original PR, do we also need to add the `kernels` package as a dependency? I see they require it in their docs [here](https://huggingface.co/kernels-community/vllm-flash-attn3)